fdbserver/MockPeekCursor.h

```diff
@@ -21,39 +21,102 @@
 #ifndef FDBSERVER_MOCK_PEEK_CURSOR
 #define FDBSERVER_MOCK_PEEK_CURSOR
 
+#include <ostream>
 #include "fdbserver/LogSystem.h"
 
+struct VersionedMessage {
+	// Every message should be able to have independent arenas so that they can be freed separately after consumed.
+	Arena arena;
+	Version version;
+	// Subversion is not used for now.
+	uint32_t sub = 0;
+	StringRef message;
+	VectorRef<Tag> tags{};
+	VersionedMessage() = default;
+	VersionedMessage(const Arena& arena, Version version, const StringRef& message, const VectorRef<Tag>& tags)
+	  : arena(arena), version(version), message(message), tags(tags) {}
+
+	friend std::ostream& operator<<(std::ostream& os, const VersionedMessage& versionedMessage) {
+		os << "VersionedMessage [version: " << versionedMessage.version << " sub: " << versionedMessage.sub
+		   << " message: " << StringRef(versionedMessage.message).toHexString() << " tags: ";
+		for (auto tag : versionedMessage.tags) {
+			os << tag.toString() << " ";
+		}
+		os << "]";
+		return os;
+	}
+};
+
+struct VersionedMessageSupplier final : ReferenceCounted<VersionedMessageSupplier> {
+	int i = 0;
+	int end;
+	VectorRef<Tag> tags{};
+	int advanceVersionsPerMutation;
+
+	Optional<VersionedMessage> get() {
```

Comment:
This method construct the message on-demand. Another way is to construct all messages in the constructor and `get()` just returns them one by one. For benchmarking, pre-constructing all messages is preferred.

Reply:
Yeah, the problem is that benchmarking can generate a large amount of messages that do not fit in memory. I guess what we can do it to generate the messages on demand but in batches. It can save the time of allocating separate arenas for every messages, so that it can be close enough to pre-constructing them. (Assuming most of the time is spent on memory allocation rather than object creation).

I think I'm not going to optimize here, considering I have to wind this PR up quickly and work on another project.